### PR TITLE
Buyankhuu/fixed queries+changed store front

### PR DESCRIPTION
**Pull request by BuyankhuuTsCAl**
[//]: # "These comments are meant for your reference. They are invisible and don't need to be deleted!"

### Description
[//]: # "Required - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."
fixed the buttons on the storefront + the shop {text} on the storefront to reflect the button selection.
### Screenshots
[//]: # "Required for frontend changes, otherwise optional but strongly recommended. Add screenshots of expected behavior - GIFs if you're feeling fancy!"

## How to review
[//]: # "Required - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?"

## Next steps
[//]: # "Optional - What's NOT in this PR, doesn't work yet, and/or still needs to be done. Note any temporary fixes in this PR that should be cleaned up later."

### Online sources
[//]: # "Optional - copy links to any tutorials or documentation that was useful to you when working on this PR"

### Related PRs
[//]: # "Optional - related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

[//]: # "This tags the project leader as a default. Feel free to change, or add on anyone who you should be in on the conversation."
🛍 CC: @EthanAuyeung

### CODE REVIEW

File: src/api/supabase/queries/user_queries.ts [hunk 1]
```diff
@@ -90,23 +91,25 @@ export async function arrayOfFavorites() {
     const { data: userProfileData, error: userProfileError } = await supabase
```
Comment (kevinjcai) [inline src/api/supabase/queries/user_queries.ts]: use the previous helper functions to do this task

File: src/api/supabase/queries/user_queries.ts [hunk 2]
```diff
@@ -39,7 +39,7 @@ export async function fetchUserByUUID(uuid: string) {
     const { data: user, error } = await supabase
```
Comment (kevinjcai) [inline src/api/supabase/queries/user_queries.ts:67]: https://github.com/calblueprint/shanti-project/blob/kevincai/new_back/src/api/supabase/queries/user_queries.ts

look at this to use auth to grab the user (still needa test a little)

File: src/api/supabase/queries/user_queries.ts [hunk 3]
```diff
@@ -62,21 +62,22 @@ export async function getUserInfo(product: Product, isFav: boolean) {
     const { data } = await supabase
```
Comment (kevinjcai) [inline src/api/supabase/queries/user_queries.ts]: edit the name of this function to something more meaningful

File: src/api/supabase/queries/user_queries.ts [hunk 4]
```diff
@@ -134,7 +137,7 @@ export async function totalNumberOfItemsInCart() {
     const { data, error } = await supabase
```
Comment (kevinjcai) [inline src/api/supabase/queries/user_queries.ts]: i would move these to the product_queries.tsx file

**Review by kevinjcai — COMMENTED** [review]
Looks pretty good! Just a few minor changes.

LMK if you have any questions or want to give over your pr

**Comment by BuyankhuuTsCAl** [comment]
> Looks pretty good! Just a few minor changes.
> 
> LMK if you have any questions or want to give over your pr

File: src/api/supabase/queries/user_queries.ts (hunk 2 above)
Comment (BuyankhuuTsCAl) [reply src/api/supabase/queries/user_queries.ts:67]: Hi Kevin, thanks for the comments! Super helpful. I don't think I wrote this function so I might pass it to you but I have resolved the rest of the comments

File: src/api/supabase/queries/product_queries.ts [hunk 5]
```diff
@@ -53,3 +53,41 @@ export async function fetchProductByID(productId: number) {
...
   }
 }
+
+/**
+ *
+ * @returns the number of items stored within the cart if there is no items then returns 0
+ */
+
+export async function totalNumberOfItemsInCart() {
+  const {
```
Comment (kevinjcai) [inline src/api/supabase/queries/product_queries.ts:63]: use the helper functions

File: src/api/supabase/queries/product_queries.ts [hunk 6]
```diff
@@ -53,3 +53,41 @@ export async function fetchProductByID(productId: number) {
...
+    data: { user },
+  } = await supabase.auth.getUser();
+
+  if (user !== null) {
+    const { data, error } = await supabase
```
Comment (kevinjcai) [inline src/api/supabase/queries/product_queries.ts:68]: use the helper function

File: src/api/supabase/queries/user_queries.ts [hunk 7]
```diff
@@ -53,107 +53,92 @@ export async function fetchUserByUUID(uuid: string) {
   }
 }
 
-export async function getUserInfo(product: Product, isFav: boolean) {
+export async function getProduct() {
```
Comment (kevinjcai) [inline src/api/supabase/queries/user_queries.ts]: can you put this in the product_queries.ts

File: src/api/supabase/queries/user_queries.ts [hunk 8]
```diff
@@ -53,107 +53,92 @@ export async function fetchUserByUUID(uuid: string) {
...
     data: { user },
   } = await supabase.auth.getUser();
 
-  if (user !== null) {
-    const { data } = await supabase
-      .from('profiles')
-      .select()
-      .eq('user_id', user.id);
+  if (user !== null && user !== undefined) {
+    const specificUser = await fetchUserByUUID(user.id);
```
Comment (kevinjcai) [inline src/api/supabase/queries/user_queries.ts]: user is defined above should be okay
